src/ui/screenshot_menu.ts

```diff
@@ -0,0 +1,806 @@
+/**
+ * @license
+ * Copyright 2024 Google Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * @file UI menu for taking screenshots from the viewer.
+ */
+
+import "#src/ui/screenshot_menu.css";
+import svg_close from "ikonate/icons/close.svg?raw";
+import svg_help from "ikonate/icons/help.svg?raw";
+import { throttle } from "lodash-es";
+import { Overlay } from "#src/overlay.js";
+import { StatusMessage } from "#src/status.js";
+import { setClipboard } from "#src/util/clipboard.js";
+import type {
+  ScreenshotLoadStatistics,
+  ScreenshotManager,
+} from "#src/util/screenshot_manager.js";
+import { MAX_RENDER_AREA_PIXELS } from "#src/util/screenshot_manager.js";
+import { ScreenshotMode } from "#src/util/trackable_screenshot_mode.js";
+import type {
+  DimensionResolutionStats,
+  PanelViewport,
+} from "#src/util/viewer_resolution_stats.js";
+import {
+  getViewerResolutionMetadata,
+  getViewerLayerResolutions,
+  getViewerPanelResolutions,
+} from "#src/util/viewer_resolution_stats.js";
+import { makeCopyButton } from "#src/widget/copy_button.js";
+import { makeIcon } from "#src/widget/icon.js";
+
+// If DEBUG_ALLOW_MENU_CLOSE is true, the menu can be closed by clicking the close button
+// Usually the user is locked into the screenshot menu until the screenshot is taken or cancelled
+// Setting this to true, and setting the SCREENSHOT_MENU_CLOSE_TIMEOUT in screenshot_manager.ts
+// to a high value can be useful for debugging canvas handling of the resize
+// Also helpful for viewing the canvas at higher resolutions
+const DEBUG_ALLOW_MENU_CLOSE = false;
+
+// For easy access to UI elements
+const PANEL_TABLE_HEADER_STRINGS = {
+  type: "Panel type",
+  pixelResolution: "Pixel resolution",
+  physicalResolution: "Physical scale",
+};
+const LAYER_TABLE_HEADER_STRINGS = {
+  name: "Layer name",
+  type: "Data type",
+  resolution: "Physical voxel resolution",
+};
+const TOOLTIPS = {
+  generalSettingsTooltip:
```

Comment:
The custom tooltips seem nice but it may make sense to remain consistent with the rest of neuroglancer and just use the title attribute.

Reply:
Sure thing, removed and used title to be consistent, thanks